components/range_cache_memory_engine/src/background.rs

```diff
@@ -604,38 +602,44 @@ impl BackgroundRunnerCore {
     ///
     /// See: [`RangeStatsManager::collect_changes_ranges`] for
     /// algorithm details.
-    fn top_regions_load_evict(&self, delete_range_scheduler: &Scheduler<BackgroundTask>) {
+    async fn top_regions_load_evict(&self, delete_range_scheduler: &Scheduler<BackgroundTask>) {
         let range_stats_manager = match &self.range_stats_manager {
             Some(m) => m,
             None => {
                 return;
             }
         };
-        if range_stats_manager.set_checking_top_regions(true) {
+        if range_stats_manager.ready_for_auto_load_and_evict() {
```

Comment:
```suggestion
        if !range_stats_manager.ready_for_auto_load_and_evict() {
```
BTW, what if IME is disabled dynamically, do we need to also check this?